### ISSUE TITLE
Sequential request handling is not practical at a large-scale

### ISSUE THREAD

**Issue by Programmix** [hydrabolt/discord.js#656]
*Oh, what have async requests come to? :(*

Sequential request handling works all right and makes bots extremely less likely to abuse rate limits. However, on a more large-scale where the bot needs to perform lots of actions per second, sequential request handling becomes less practical. I think that it works great for smaller-scale bots, however. A separate, optional request handler should be implemented that can deal with multiple requests being sent at once (and thus should also have an RPS lock.)

**Comment by hydrabolt** [comment]
9.2 / 9.3 will feature a `Batch`/`Bucket` rate limit handler so that requests can run concurrently, as is the case in <9.x

**Comment by hydrabolt** [comment]
I've addressed this in e75dd2638c2e479b5756b38986fe980582beda6d, you can now use burst request mode to send requests as soon as they come through until the requester runs out of remaining requests or gets rate-limited. After a specified amount of time, the requester will begin to process its queue again. 

While this _is_ experimental, it seems to work just fine! You can use `burst` mode like so:

```js
const client = new Discord.Client({ api_request_method: 'burst' });
```
